backend/qemu.pm

```diff
@@ -814,8 +823,22 @@ sub start_qemu {
     sp('only-migratable') if $self->can_handle({function => 'snapshots', no_warn => 1});
     sp('chardev', 'ringbuf,id=serial0,logfile=serial0,logappend=on');
     sp('serial',  'chardev:serial0');
-    my $soundhw = $vars->{QEMU_SOUNDHW} // 'hda';
-    sp('soundhw', $soundhw);
+
+    # poo#66667: Since qemu 4.2, -audiodev is required to record sounds, as we need an audiodev id for 'wavcapture'
+    if (version->declare($vars->{QEMU_VERSION}) lt version->declare(4.2)) {
```

Comment:
This is only the 2nd repetition of this condition but maybe it makes sense to have a separate function like `requires_audiodev` instead of hard-coding the version number on multiple places.